src/components/ui/dashboard/join-coaching-session.tsx

```diff
@@ -0,0 +1,186 @@
+    setOrganizationId(value);
+    if (value) {
+      fetchOrganization(value).then(([organization]) => {
+        setOrganization(organization);
+      });
+    }
+  };
+
+  const handleRelationshipSelection = (selectedRelationship: Id) => {
+    setRelationshipId(selectedRelationship);
+    if (selectedRelationship && organizationId) {
+      fetchCoachingRelationshipWithUserNames(
+        organizationId,
+        selectedRelationship
+      ).then((relationship) => {
+        setCoachingRelationship(relationship);
+      });
+    }
+  };
+
+  const handleSessionSelection = (selectedSession: Id) => {
+    if (selectedSession && relationshipId) {
+      fetchCoachingSessions(relationshipId).then(([sessions]) => {
+        const theSession = sessions.find(
+          (session) => session.id === selectedSession
+        );
+        if (theSession) {
+          setCoachingSession(theSession);
+          setCoachingSessionId(theSession.id);
+        }
+      });
+    }
+  };
+
+  return (
+    <Card className="w-[300px]">
+      <CardHeader>
+        <CardTitle>Join a Coaching Session</CardTitle>
+      </CardHeader>
+      <CardContent className="grid gap-6">
+        <div className="grid gap-2">
+          <Label htmlFor="organization-selector">Organization</Label>
+          <DynamicApiSelect<Organization>
+            url="/organizations"
+            params={{ userId }}
+            onChange={handleOrganizationSelection}
+            placeholder={orgPlaceholder}
+            getOptionLabel={(org) => org.name}
+            getOptionValue={(org) => org.id.toString()}
+            elementId="organization-selector"
+          />
+        </div>
+        {organizationId && (
+          <div className="grid gap-2">
+            <Label htmlFor="relationship-selector">Relationship</Label>
+            <DynamicApiSelect<CoachingRelationshipWithUserNames>
+              url={`/organizations/${organizationId}/coaching_relationships`}
+              params={{ organizationId }}
+              onChange={handleRelationshipSelection}
+              placeholder={relPlaceholder}
+              getOptionLabel={(relationship) =>
+                `${relationship.coach_first_name} ${relationship.coach_last_name} -> ${relationship.coachee_first_name} ${relationship.coachee_last_name}`
+              }
+              getOptionValue={(relationship) => relationship.id.toString()}
+              elementId="relationship-selector"
+            />
+          </div>
+        )}
+        {relationshipId && (
+          <div className="grid gap-2">
+            <Label htmlFor="session-selector">Coaching Session</Label>
+            <DynamicApiSelect<CoachingSession>
+              url="/coaching_sessions"
+              params={{
+                coaching_relationship_id: relationshipId,
+                from_date: FROM_DATE,
+                to_Date: TO_DATE,
+              }}
+              onChange={handleSessionSelection}
+              placeholder={sessionPlaceholder}
```

Comment:
When you select a session and join, and then return to the dashboard page, it'll show the date of the selected session but it's in a raw datetimestampz format instead of the same format used in the drop down list in the selector.

Reply:
Yup! See slack DM! That was the last bit driving me mad 😠. Not an ideal solution but I'll add a reformatter to the option content itself. Let me know if you can think of something else in the sorting part of the dynamic select component.

Reply:
Turned out to be value of the date in the state tree. 😄